projects/output-and-error-handling/client/main.go

```diff
@@ -0,0 +1,184 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+// Comments ----------
+// I write a lot of comments, it has helped me the entire course to learn, understand and remember
+// They can all be removed at the end to clean it up
+
+// Error Messages ----------
+// The Errors are deliberately numbered [x] so I could follow them up the call stack
+// They can be replaced with "user friendly" messages later
+
+func parseResponseBody(response *http.Response) (string, error) {
+	// try to read the response body
+	responseBody, err := io.ReadAll(response.Body)
+
+	// handle any error immediately
+	if err != nil {
+		return "", fmt.Errorf("[2] failed to read response body : %w", err)
+	}
+
+	// use the defer keyword early to close the body regardless
+	defer response.Body.Close()
+
+	return string(responseBody), nil
+}
+
+// returned integer is the retry duration in seconds
+func parseRetryAfterHeader(retryAfterHeaderString string) (int, error) {
```

Comment:
You make it very clear that this will return `int` duration in seconds and it's perfectly fine, but do you think there's another type that you could return instead of `int`? 

hint: this will save you some conversions

Reply:
I went with time duration as the type because that makes the most sense(?) as in its describing the purpose the value
But i have no idea if you meant that (seems like you were hinting at string?)
As a benefit it also means the retry works in decimal increments

Reply:
No I meant `time.Duration`